services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/RouterBolt.java

```diff
@@ -35,21 +36,48 @@
 import org.openkilda.messaging.nbtopology.request.SwitchesBaseRequest;
 import org.openkilda.wfm.AbstractBolt;
 import org.openkilda.wfm.error.PipelineException;
+import org.openkilda.wfm.share.utils.KeyProvider;
 import org.openkilda.wfm.topology.nbworker.StreamType;
+import org.openkilda.wfm.topology.nbworker.commands.RemoveKeyRouterCommand;
 import org.openkilda.wfm.topology.utils.MessageKafkaTranslator;
 
 import org.apache.storm.topology.OutputFieldsDeclarer;
 import org.apache.storm.tuple.Fields;
 import org.apache.storm.tuple.Tuple;
 import org.apache.storm.tuple.Values;
 
+import java.util.HashSet;
+import java.util.Set;
+
 public class RouterBolt extends AbstractBolt {
+    public static final String INCOME_STREAM = "router.bolt.stream";
+    private Set<String> keys = new HashSet<>();
 
     @Override
     protected void handleInput(Tuple input) throws PipelineException {
         String key = input.getStringByField(MessageKafkaTranslator.FIELD_ID_KEY);
         Message message = pullValue(input, FIELD_ID_PAYLOAD, Message.class);
 
+        if (keys.contains(KeyProvider.getParentKey(key))) {
```

Comment:
Why can't we route a request based on type/class?

Reply:
it looks like this is already resolved by field-grouping on the storm level, try to remove the keys at all

Reply:
Fixed.